scripts/tests/sanitycheck/test_reporting_testsuite.py

```diff
@@ -0,0 +1,151 @@
+    """ Testing csv_report function of Testsuite class in sanitycheck
+	Test 1: Assert the csv_report isnt empty after execution of csv_report function
+	Test 2: Assert on the columns and values of the generated csv_report"""
+    class_testsuite.instances = instances_fixture
+    filename = tmpdir.mkdir("test_csv").join("sanitycheck_csv_report.csv")
+    class_testsuite.csv_report(filename)
+    csvreport_df = pd.read_csv(filename)
+    csvreport_df = csvreport_df.where(pd.notnull(csvreport_df), None)
+    assert os.path.exists(filename)
+    assert not csvreport_df.empty
+
+    mydict = {'test': [], 'arch' : [], 'platform' : [], 'status': [],
+              'extra_args': [], 'handler': [], 'handler_time': [],
+              'ram_size': [], 'rom_size': []}
+    assert set(mydict.keys()) == set(csvreport_df.columns)
+
+    for instance in class_testsuite.instances.values():
+        mydict["test"].append(instance.testcase.name)
+        mydict["arch"].append(instance.platform.arch)
+        mydict["platform"].append(instance.platform.name)
+        mydict["status"].append(instance.status)
+        args = " ".join(instance.testcase.extra_args) if instance.testcase.extra_args else None
+        mydict["extra_args"].append(args)
+        mydict["handler"].append(instance.platform.simulation)
+        mydict["handler_time"].append(instance.metrics.get("handler_time", None))
+        mydict["ram_size"].append(instance.metrics.get("ram_size", 0))
+        mydict["rom_size"].append(instance.metrics.get("rom_size", 0))
+
+    for i in range(len(mydict)):
+        assert set(csvreport_df[list(mydict.keys())[i]]) == set(list(mydict.values())[i])
+
+def test_xunit_report(class_testsuite, test_data, instances_fixture, tmpdir_factory):
+    """ Testing xunit_report function of Testsuite class in sanitycheck
+	Test 1: Assert sanitycheck.xml file exists after execution of xunit_report function
+	Test 2: Assert on fails, passes, skips, errors values
+	Test 3: Assert on the tree structure of sanitycheck.xml file"""
+    class_testsuite.instances = instances_fixture
+    class_testsuite.outdir = tmpdir_factory.mktemp("sanity_out_demo")
+    inst1 = class_testsuite.instances.get("demo_board_2/scripts/tests/sanitycheck/test_data/testcases/tests/test_a/test_a.check_1")
+    inst2 = class_testsuite.instances.get("demo_board_2/scripts/tests/sanitycheck/test_data/testcases/tests/test_a/test_a.check_2")
+    inst1.status = "failed"
+    inst2.status = "skipped"
+    filename = test_data + "sanitycheck.xml"
+    class_testsuite.xunit_report(filename)
+    assert os.path.exists(filename)
+
+    filesize = os.path.getsize(filename)
+    assert filesize != 0
+
+    fails, passes, errors, skips = 0, 0, 0, 0
+    for instance in class_testsuite.instances.values():
+        if instance.status in ["failed", "timeout"]:
+            if instance.reason in ['build_error', 'handler_crash']:
+                errors += 1
+            else:
+                fails += 1
```

Comment:
fails here will just count timeouts right ? I'd change the variable name to be clear, because fails in this context should be timeouts + errors (build errors and crashes)

Reply:
This is using the same logic as is used in sanitylib  script in xunit_report function for calculating the fails and errors. I think the fails also account for build failures.

Reply:
that is fine, but that is a bad name for this variable. Since fails just count timeouts and you have another counter to build errors and crashes. But this is minor though.

Reply:
Just want to be consistent here with the terminology used in sanitylib script.  In future if that changes, will fix this. Thanks for bringing that up.

Reply:
Seems like...this is being handled in the PR : https://github.com/zephyrproject-rtos/zephyr/pull/26543. The terminology hasnt changed but the now build errors will be reported as errors, not test failures. Will not need to address this i think.